roles/postgresql_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,123 @@
+  become_user: postgres
+  register: postgresql_upgrade_tables_with_oids_ddl
+  when: postgresql_upgrade_check_for_oids
+
+- name: Run ALTER statements for tables with OIDS
+  community.postgresql.postgresql_query:
+    db: "{{ postgressql_upgrade_database }}"
+    query: "{{ item.query }}"
+  loop: "{{ postgresql_upgrade_tables_with_oids_ddl.query_result }}"
+  become: true
+  become_user: postgres
+  when: postgresql_upgrade_check_for_oids
+
+- name: Back up PostgreSQL after OIDS edit
+  ansible.builtin.command: "{{ postgreql_upgrade_backup_script }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+  when: postgresql_upgrade_check_for_oids
+
+- name: Disable previous version of PostgreSQL - {{ postgreql_upgrade_current_version }}
+  ansible.builtin.service:
+    name: "postgresql-{{ postgreql_upgrade_current_version }}"
+    state: stopped
+    enabled: false
+
+- name: Install new version of PostgreSQL - {{ postgreql_upgrade_new_version }}
+  ansible.builtin.include_role:
+    name: mirsg.infrastructure.postgresql
+  vars:
+    postgresql_version: "{{ postgreql_upgrade_new_version }}" # noqa var-naming[no-role-prefix]
+    postgresql_create_database: false
+
+- name: Ensure postgres is not running during configuration change
+  ansible.builtin.service:
+    name: "postgresql-{{ postgreql_upgrade_new_version }}"
+    state: stopped
+
+- name: Remove any previous upgrade output
+  ansible.builtin.file:
+    path: "{{ postgresql_upgrade_scripts_dir }}"
+    state: absent
+
+- name: >-
+    Ensure directory for upgrade scripts exists -
+    {{ postgresql_upgrade_scripts_dir }}
+  ansible.builtin.file:
+    path: "{{ postgresql_upgrade_scripts_dir }}"
+    owner: "{{ postgresql_upgrade_postgresql_owner }}"
+    group: "{{ postgresql_upgrade_postgresql_group }}"
+    state: directory
+    mode: "0700"
+
+- name: Run PostgreSQL upgrade
+  ansible.builtin.command:
+    cmd: >
+      /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin/pg_upgrade -d
+      /var/lib/pgsql/{{ postgreql_upgrade_current_version }}/data -D
+      /var/lib/pgsql/{{ postgreql_upgrade_new_version }}/data -b
+      /usr/pgsql-{{ postgreql_upgrade_current_version }}/bin -B
+      /usr/pgsql-{{ postgreql_upgrade_new_version }}/bin -p 10094 -P 5432
+    chdir: "{{ postgresql_upgrade_scripts_dir }}"
+  become: true
+  become_user: postgres
+  changed_when: false
+
+- name: Ensure new postgres is running - {{ postgreql_upgrade_new_version }}
+  ansible.builtin.service:
+    name: "postgresql-{{ postgreql_upgrade_new_version }}"
+    state: started
+    enabled: true
+
+# Note: The script(s) to run could change for different upgrades; you may need to modify this command each time
+- name: Run scripts produced by upgrade command (NOTE - these could change for each upgrade)
+  ansible.builtin.command:
+    cmd: "{{ postgresql_upgrade_generated_script }}"
+    chdir: "{{ postgresql_upgrade_scripts_dir }}"
+  become: true
+  become_user: postgres
+  changed_when: false
```

Comment:
is the comment about changing the command each time still accurate, or is that handled by have defined `postgresql_upgrade_scripts_dir` and `postgresql_upgrade_generated_script`?

Reply:
Sort of. The upgrade to `14` generated a `delete_old_cluster.sh` script rather than `analyze_new_cluster.sh` as it was for an upgrade to `12`. It is handled by the variables but those variables will potentially need overriding, so the sentiment of the comment is right. It should be described in the README which I've forgotten to add though...